core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -1070,6 +1075,13 @@ public void packetError(PacketErrorEvent event) {
         downstream.connect(false, loginEvent.transferring());
     }
 
+    public void updateTickData(float tickRate, boolean frozen) {
+        tickThread.cancel(true);
```

Comment:
```suggestion
    public void updateTickingState(float tickRate, boolean frozen) {
        tickThread.cancel(true);
```
also please move to immediately above the tick() method

Reply:
Fixed in latest commit.